src/h5web/vis-packs/core/hooks.ts

```diff
@@ -110,16 +110,32 @@ export function useFrameRendering(): void {
   });
 }
 
+export function useVisSize(ratio: number | undefined): Size {
+  const { width, height } = useThree((state) => state.size);
+
+  if (!ratio) {
+    return { width, height };
+  }
+
+  const canvasRatio = width / height;
+  const mismatch = canvasRatio / ratio;
+
+  return {
+    width: mismatch > 1 ? height * ratio : width,
+    height: mismatch > 1 ? height : width / ratio,
+  };
```

Comment:
What about:
```js
if(canvasRatio > ratio) {
  return {width: height*ratio, height}
}

return { width, height: width / ratio }
```

Reply:
I think the reason I like the single return is that the two branches have equal value, whereas the early return makes it look like `canvasRatio > ratio` is a special case.

Reply:
Though I'm happy to replace `mismatch > 1` with `canvasRatio > ratioToRespect`

Reply:
Good old forgotten pal `else` to the rescue:

```js
if(canvasRatio > ratio) {
  return {width: height*ratio, height}
}
else {
  return { width, height: width / ratio }
}
```

But I think that it prevented by our linting rules. I liked the fact that the `if` quickly shows that only one part of the size changes but I am not deeply against the current writing with ternaries.